Classes/Message/MessageMetadata.php

```diff
@@ -59,25 +58,28 @@ public function getProperties(): array
      * @param string $name
      * @param mixed $value
      * @return MessageMetadata
-     * @throws MessageMetadataException
      */
-    public function add(string $name, $value)
+    public function withProperty(string $name, $value)
```

Comment:
you could have added some nice method description while you were at it ... but we can as well do that in a later pull request.